### PR TITLE
Add `TeleconsultStatusSheet` #1

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/1623/create-teleconsultstatusdialog

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt [hunk 1]
```diff
@@ -6,13 +6,17 @@ import com.spotify.mobius.test.InitSpec
...
 
 class TeleconsultStatusInitTest {
 
   @Test
   fun `when screen is created, then prefill teleconsult status if it exists`() {
     val initSpec = InitSpec(TeleconsultStatusInit())
-    val model = TeleconsultStatusModel.create(TeleconsultStatus.Yes)
+    val model = TeleconsultStatusModel.create(
+        teleconsultRecordId = UUID.fromString("60af7f4a-b61f-40ae-a2b9-9f5273a70d4e"),
+        teleconsultStatus = TeleconsultStatus.Yes
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt:18]: Creating a model in the tests should follow the process they do in production. In production, we will never have a teleconsult status ready when creating the model and we'll end up constructing it via the events. We should follow the same flow here.

```kotlin
val model = TeleconsultStatusModel
  .create(UUID.fromString())
  .teleconsultStatusLoaded()
```

File: app/src/main/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusModel.kt [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 
 data class TeleconsultStatusModel(
+    val teleconsultRecordId: UUID,
     val teleconsultStatus: TeleconsultStatus?
 ) {
 
   companion object {
 
-    fun create(teleconsultStatus: TeleconsultStatus?) = TeleconsultStatusModel(
+    fun create(teleconsultRecordId: UUID, teleconsultStatus: TeleconsultStatus?) = TeleconsultStatusModel(
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusModel.kt:13]: We've mixed two changes in this commit:

- Adding the teleconsult record ID as a property of the model
- Handling the done click event

These should be two separate commits for easier review.

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]

File: app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt:18]: If user selects the still waiting status, won't we have to reselect it next time the sheet is shown? 🤔 . That was the only reason why I included default teleconsult status option, or else we can remove prefill effect as well.

File: app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusUpdateTest.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.spotify.mobius.test.UpdateSpec.assertThatNext
+import org.junit.Test
+import org.simple.clinic.teleconsultlog.teleconsultrecord.TeleconsultStatus
+
+class TeleconsultStatusUpdateTest {
+
+  @Test
+  fun `when teleconsult status is changed, the update the model`() {
+    val updateSpec = UpdateSpec(TeleconsultStatusUpdate())
+    val model = TeleconsultStatusModel.create(null)
```
[the remark this answers is not in the document]
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusUpdateTest.kt]: The initial idea behind this was to allow setting default `TeleconsultStatus`, but if we don't have to set default option I can remove this and prefill teleconsult status.

File: app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultUiRendererTest.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import org.junit.Test
+import org.simple.clinic.teleconsultlog.teleconsultrecord.TeleconsultStatus
+
+class TeleconsultUiRendererTest {
+
+  @Test
+  fun `when teleconsult status is not null, then enable done button`() {
```
[the remark this answers is not in the document]
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultUiRendererTest.kt:13]: By default, we are not doing anything with model data. We have the button disabled in the layout initially, user cannot deselect the radio buttons anyway, so it didn't make sense to have a disable button in ui renderer WDYT?

File: app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt:18]: Isn't that taken care of by the prefill status effect? When the effect is triggered, it prefills the UI which triggers a checked change event which then updates the model again.

File: app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultUiRendererTest.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultUiRendererTest.kt:13]: Ah, makes sense.

File: app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt:18]: NVM, turns out we don't have to prefill the status in sheet. Confirmed with Pragati, sorry 😅 

Will remove the prefill effect as well default status in the factory method.

Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt:18]: Correct me if I am wrong, State restoration should automatically work with the RadioGroup so we don't manually have to prefill it in those scenarios as well right?

Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/summary/teleconsultation/status/TeleconsultStatusInitTest.kt:18]: Yup, that should be handled by the system

**Review by vinaysshenoy — APPROVED** [review]
